R/plot.R

```diff
@@ -0,0 +1,111 @@
+#' @title Simulate biomarker kinetics predicted by the given biokinetics priors
+#' and optionally compare to a dataset.
+#' @export
+#' @description Simulate trajectories by drawing random samples from the given
+#' priors for each parameter in the biokinetics model.
+#' @return A ggplot2 object.
+#' @param x A named list of type 'biokinetics_priors'.
+#' @param \dots Further arguments passed to the method.
+#' @param tmax Integer. The number of time points in each simulated trajectory. Default 150.
+#' @param n_draws Integer. The number of trajectories to simulate. Default 2000.
+#' @param data Optional data.frame with columns time_since_last_exp and value. The raw data to compare to.
+plot.biokinetics_priors <- function(x,
+                                    ...,
+                                    tmax = 150,
+                                    n_draws = 2000,
+                                    data = NULL) {
+
+  # Declare variables to suppress notes when compiling package
+  # https://github.com/Rdatatable/data.table/issues/850#issuecomment-259466153
+  t0 <- tp <- ts <- m1 <- m2 <- m3 <- NULL
+  time_since_last_exp <- me <- lo <- hi <- value <- mu <- NULL
+
+  if (!is.null(data)) {
+    validate_required_cols(data, c("time_since_last_exp", "value"))
+  }
+  params <- data.table(
+    t0 = stats::rnorm(n_draws, x$mu_t0, x$sigma_t0), # titre value at t0
+    tp = stats::rnorm(n_draws, x$mu_tp, x$sigma_tp), # time of peak
+    ts = stats::rnorm(n_draws, x$mu_ts, x$sigma_ts), # time of set point
+    m1 = stats::rnorm(n_draws, x$mu_m1, x$sigma_m1), # gradient 1
+    m2 = stats::rnorm(n_draws, x$mu_m2, x$sigma_m2), # gradient 2
+    m3 = stats::rnorm(n_draws, x$mu_m3, x$sigma_m3) # gradient 3
+  )
+
+  times <- data.table(t = 1:tmax)
+  params_and_times <- times[, as.list(params), by = times]
+
+  params_and_times[, mu := biokinetics_simulate_trajectory(t, t0, tp, ts, m1, m2, m3),
+                     by = c("t", "t0", "tp", "ts", "m1", "m2", "m3")]
+
+  summary <- params_and_times[, .(me = stats::quantile(mu, 0.5, names = FALSE),
+                                  lo = stats::quantile(mu, 0.025, names = FALSE),
+                                  hi = stats::quantile(mu, 0.975, names = FALSE)), by = t]
+
+  plot <- ggplot(summary) +
+    geom_line(aes(x = t, y = me)) +
+    geom_ribbon(aes(x = t, ymin = lo, ymax = hi), alpha = 0.5)
+
+  if (!is.null(data)) {
+    plot <- plot + geom_point(data = data, aes(x = time_since_last_exp, y = value))
+  }
+  plot
+}
+
+plot_data <- function(data, covariates) {
```

Comment:
Would maybe add a more descriptive name for this, maybe plot_sero_data ? Also is there the option to plot without covariates (what is covariate = NULL)?

Reply:
`covariates=NULL` will work yeah, it will just facet by `titre_type`. I'll add a default :+1:

Reply:
Ok great